### PR TITLE
initial checkin - sqldb interface

### PR DESCRIPTION
**Pull request by stevelr**
Interface for an sqldb provider wasmcloud:sqldb

Signed-off-by: stevelr <steve@cosmonic.com>

### CODE REVIEW

File: sqldb/rust/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+[package]
+name = "wasmcloud-interface-sqldb"
+version = "0.1.0"
+description = "interface for actors to issue http/https requests (wasmcloud:httpclient)"
```
Comment (autodidaddict) [inline sqldb/rust/Cargo.toml:4]: Suuuuper nitpicky, but fields like this that are published on crates should start with an uppercase letter.

File: sqldb/rust/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+repository = "https://github.com/wasmcloud/interfaces"
+homepage = "https://github.com/wasmcloud/wasmcloud"
+keywords = ["wasmcloud","wasm","actor","webassembly","capability"]
+categories = [ "api-bindings","network-programming","wasm" ]
+authors = [ "wasmcloud Team" ]
+edition = "2018"
+license = "Apache-2.0"
```
Comment (autodidaddict) [inline sqldb/rust/Cargo.toml:11]: should also specify the `readme` field to indicate the location of the `README` to display on crates dot io.

File: sqldb/rust/src/sqldb.rs [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+pub struct Column {
+    /// Column name in the result
+    #[serde(default)]
+    pub name: String,
+    /// column ordinal
+    pub ordinal: u32,
+    /// Data type of the column
+    #[serde(default)]
+    pub ty: String,
```
Comment (autodidaddict) [inline sqldb/rust/src/sqldb.rs:29]: again, another super tiny nitpick, but historically I've seen folks either go with `typ` or `datatype` when they're unable to use the word `type` as a field name.

File: sqldb/sqldb.smithy [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    @required
+    rows: Blob,
+}
+
+structure SqlDbError {
+
+    /// Type of error.
+    /// The list of error codes below may be expanded in the future
+    /// to provide finer-granularity failure information
+    @enum([
```
Comment (autodidaddict) [inline sqldb/sqldb.smithy:99]: Ooh I like that we're using the smithy-based enum here. Sets a good example for future model designers.

**Review by autodidaddict — CHANGES_REQUESTED** [review]
Just requesting a couple of _really_ tiny changes to finish up the shine prior to crate publication.

File: sqldb/sqldb.smithy (hunk 4 above)
Comment (stevelr) [reply sqldb/sqldb.smithy:99]: There's a backlog issue to add enums to the code generator if they're specified in smithy. Currently you would call this a "documentation trait" - i.e., communication to the developer but the constraint is not technically enforced.

File: sqldb/rust/src/sqldb.rs (hunk 3 above)
Comment (stevelr) [reply sqldb/rust/src/sqldb.rs:29]: changed to dbType (see comment in slack)

**Review by autodidaddict — APPROVED** [review]

**Pull request merged**